frontend/pages/software/ManageSoftwarePage/components/ManageAutomationsModal/ManageAutomationsModal.tsx

```diff
@@ -71,15 +129,163 @@ const ManageAutomationsModal = ({
       ...newErrors,
     });
 
-    // URL validation only needed if software automation is checked
-    if (valid || !softwareAutomationsEnabled) {
-      onCreateWebhookSubmit({
-        destination_url,
-        enable_vulnerabilities_webhook: softwareAutomationsEnabled,
-      });
+    // Original config keys for software automation (webhook_settings, integrations)
+    const configSoftwareAutomations: ISoftwareAutomations = {
+      webhook_settings: {
+        vulnerabilities_webhook: {
+          destination_url,
+          enable_vulnerabilities_webhook: softwareVulnerabilityWebhookEnabled,
+        },
+      },
+      integrations: {
+        jira: integrations || [],
+      },
+    };
 
-      onReturnToApp();
+    if (!softwareAutomationsEnabled) {
+      // set enable_vulnerabilities_webhook to false and all jira.enable_software_vulnerabilities to false
+      configSoftwareAutomations.webhook_settings.vulnerabilities_webhook.enable_vulnerabilities_webhook = false;
+      const disableAllJira = configSoftwareAutomations.integrations.jira.map(
+        (integration) => {
+          return { ...integration, enable_software_vulnerabilities: false };
+        }
+      );
+      configSoftwareAutomations.integrations.jira = disableAllJira;
+    } else if (!jiraEnabled) {
+      if (valid) {
+        // set enable_vulnerabilities_webhook to true and all jira.enable_software_vulnerabilities to false
+        configSoftwareAutomations.webhook_settings.vulnerabilities_webhook.enable_vulnerabilities_webhook = true;
+        const disableAllJira = configSoftwareAutomations.integrations.jira.map(
+          (integration) => {
+            return {
+              ...integration,
+              enable_software_vulnerabilities: false,
+            };
+          }
+        );
+        configSoftwareAutomations.integrations.jira = disableAllJira;
+      } else {
+        return; // do not send request to API for webhook automation if url is !valid
+      }
+    } else {
+      // set enable_vulnerabilities_webhook to false and all jira.enable_software_vulnerabilities to false
+      // except the one jira integration selected
+      configSoftwareAutomations.webhook_settings.vulnerabilities_webhook.enable_vulnerabilities_webhook = false;
+      const enableSelectedJiraIntegrationOnly = configSoftwareAutomations.integrations.jira.map(
+        (integration, index) => {
+          return {
+            ...integration,
+            enable_software_vulnerabilities:
+              index === selectedIntegration?.integrationIndex,
+          };
+        }
+      );
+      configSoftwareAutomations.integrations.jira = enableSelectedJiraIntegrationOnly;
```

Comment:
I'm having a hard time reading this block. I think it's partially the long chain of config selectors, but also the if/else conditions don't seem particularly related. You could break up the if/if else/else into their own blocks, which would help readability. 

By wrapping the block in a self-invoking function you could return when you want the function to stop running. That'll mean the last bit doesn't have to be in a condition at all. The last bit is the default behavior, and the if statements catch various conditions and short-circuit (`return`) the function. That saves you the final condition. 

What about something like this?

```javascript
(() => {
  if (!softwareAutomationsEnabled) {
    // set enable_vulnerabilities_webhook to false and all jira.enable_software_vulnerabilities to false
    ...
    return;
  }

  if (!jiraEnabled) {
    // set enable_vulnerabilities_webhook to true and all jira.enable_software_vulnerabilities to false
    ...
    return;
  }

  // set enable_vulnerabilities_webhook to false and all jira.enable_software_vulnerabilities to false
  // except the one jira integration selected
  ...
});

```